### PR TITLE
91 messages into firebase jdt

### PR DESCRIPTION
**Pull request by JulienDiTria**
make chats and messages interaction with firebase

### CODE REVIEW

File: app/src/main/java/com/github/sdp/tarjetakuna/database/ChatsRTDB.kt [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    }
+
+    /**
+     * Get a chat from the database.
+     */
+    fun getChatFromDatabase(chatUID: String): CompletableFuture<Chat> {
+        val future = CompletableFuture<Chat>()
+        getChat(chatUID).thenAccept {
+            val chat = DBChat.fromDBChat(it)
+            messagesRTDB.getMessagesFromDatabase(it.messages).thenAccept { it2 ->
```
Comment (Bjolidon) [inline app/src/main/java/com/github/sdp/tarjetakuna/database/ChatsRTDB.kt:84]: Does this work with the nested thenAccept? I personally had some problems when trying to implement something similar in UserRTDB and had to use thenCompose.

File: firebase_emulator_settings.txt
Comment (Bjolidon) [inline firebase_emulator_settings.txt:1]: Thank you for these explanations! I now understand why I struggled to make tests pass for UserRTDB...

**Review by Bjolidon — APPROVED** [review]
Perhaps we should consider including messaging in the local database. Apart from that, the code looks very good and increases test coverage. Good job!

File: app/src/main/java/com/github/sdp/tarjetakuna/database/DBChat.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package com.github.sdp.tarjetakuna.database
+
+import com.github.sdp.tarjetakuna.model.Chat
+import com.github.sdp.tarjetakuna.model.Message
+import com.github.sdp.tarjetakuna.model.User
+import java.util.Date
+
+data class DBChat(
+    /**
```
[the remark this answers is not in the document]
Comment (JulienDiTria) [reply app/src/main/java/com/github/sdp/tarjetakuna/database/DBChat.kt:9]: yes, that would be todo for next sprint i think !

File: app/src/main/java/com/github/sdp/tarjetakuna/database/DBMessage.kt
[the remark this answers is not in the document]
Comment (JulienDiTria) [reply app/src/main/java/com/github/sdp/tarjetakuna/database/DBMessage.kt:1]: same here

File: app/src/main/java/com/github/sdp/tarjetakuna/database/ChatsRTDB.kt (hunk 1 above)
Comment (JulienDiTria) [reply app/src/main/java/com/github/sdp/tarjetakuna/database/ChatsRTDB.kt:84]: yeah it depends what you return but it worked i think, let me double check

File: firebase_emulator_settings.txt
Comment (JulienDiTria) [reply firebase_emulator_settings.txt:1]: sure thing

File: app/src/main/java/com/github/sdp/tarjetakuna/database/ChatsRTDB.kt (hunk 1 above)
Comment (JulienDiTria) [reply app/src/main/java/com/github/sdp/tarjetakuna/database/ChatsRTDB.kt:84]: just made some debugging test and yes it works !

**Pull request merged**